bundle/config/mutator/normalize_paths.go

```diff
@@ -0,0 +1,102 @@
+package mutator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	pathlib "path"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config/mutator/paths"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type normalizePaths struct{}
+
+func (a normalizePaths) Name() string {
+	return "NormalizePaths"
+}
+
+// NormalizePaths is applied to resources declared in YAML to translate
+// paths that are relative to YAML file locations to paths that are relative
+// to the bundle root.
+//
+// Pre-conditions:
+//   - Resources and artifacts have resolved all variables where relative paths are
+//     used (including complex variables).
+//   - Each path value is a string and has a location. Locations are absolute paths.
```

Comment:
Not necessarily true, but if they are relative, then the bundle root path will also be relative, and the relative computation between them will work out fine.

Reply:
Do you mean that "Locations are absolute paths." is not necessarily true? If we want to support non-absolute locations we should add tests for it, not 100% sure it works.